tests/framework/framework.go

```diff
@@ -0,0 +1,239 @@
+package framework
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/glog"
+	"k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	cdiClientset "kubevirt.io/containerized-data-importer/pkg/client/clientset/versioned"
+)
+
+const (
+	NsCreateTime = 30 * time.Second
+	NsDeleteTime = 5 * time.Minute
+)
+
+// run-time flags
+var (
+	kubectlPath  *string
+	ocPath	     *string
+	cdiInstallNs *string
+	kubeConfig   *string
+	master       *string
+)
+
+// Framework supports common operations used by functional/e2e tests. It holds the k8s and cdi clients,
+// a generated unique namespace, run-time flags, and more fields will be added over time as cdi e2e
+// evolves. Global BeforeEach and AfterEach are called in the Framework constructor.
+type Framework struct {
+	// prefix for generated namespace
+	NsPrefix string
+	//  k8s client
+	K8sClient *kubernetes.Clientset
+	// cdi client
+	CdiClient *cdiClientset.Clientset
+	// generated/unique ns per test
+	Namespace *v1.Namespace
+	// generated/unique secondary ns for testing across namespaces (eg. clone tests)
+	Namespace2 *v1.Namespace // note: not instantiated in NewFramework
+	// list of ns to delete beyond the generated ns
+	namespacesToDelete []*v1.Namespace
+	// test run-time flags
+	KubectlPath  string
+	OcPath	     string
+	CdiInstallNs string
+	KubeConfig   string
+	Master       string
+}
+
+// TODO: look into k8s' SynchronizedBeforeSuite() and SynchronizedAfterSuite() code and their general
+//       purpose test/e2e/framework/cleanup.go function support.
+
+// initialize run-time flags
+func init() {
+	kubectlPath = flag.String("kubectl-path", "kubectl", "Set path to kubectl binary")
+	ocPath = flag.String("oc-path", "oc", "Set path to oc binary")
+	cdiInstallNs = flag.String("installed-namespace", "kube-system", "Set the namespace CDI is installed in")
+	kubeConfig = flag.String("kubeconfig", "/var/run/kubernetes/admin.kubeconfig", "absolute path to the kubeconfig file")
+	master = flag.String("master", "https://localhost:6443", "master url:port")
```

Comment:
Can you leave the default for master as a blank especially since you can set it with #352 now

Reply:
ack. yes in my next commit